qgroundcontrol.pro

```diff
@@ -86,17 +86,34 @@ exists(user_config.pri):infile(user_config.pri, CONFIG) {
 # is present. It's useful to run "regular" builds to make sure you didn't
 # break anything.
 
+CUSTOM_DIR = $$PWD/custom
+
 contains (CONFIG, QGC_DISABLE_CUSTOM_BUILD) {
     message("Disable custom build override")
 } else {
-    exists($$PWD/custom/custom.pri) {
-        message("Found custom build")
+    CUSTOM_PRI = $$PWD/custom/custom.pri
+    exists($$CUSTOM_PRI) {
+        message("* Found custom build *")
         CONFIG  += CustomBuild
         DEFINES += QGC_CUSTOM_BUILD
         # custom.pri must define:
         # CUSTOMCLASS  = YourIQGCCorePluginDerivation
         # CUSTOMHEADER = \"\\\"YourIQGCCorePluginDerivation.h\\\"\"
-        include($$PWD/custom/custom.pri)
+        include($$CUSTOM_PRI)
+    } else {
+        CUSTOM_PRI = $$PWD/../custom/custom.pri
```

Comment:
I would change the out of tree dir name to be something less generic like 'qgc-custombuild' to be safer against collisions.